test/services/katello/pulp/rpm_test.rb

```diff
@@ -38,6 +38,31 @@ def test_update_model_is_idempotent
       end
     end
 
+    class RpmTestWithProxy < ActiveSupport::TestCase
+      include RepositorySupport
+
+      def setup
+        User.current = users(:admin)
+
+        @default_proxy = FactoryBot.create(:http_proxy, name: 'best proxy')
+        Setting.find_by(name: 'content_default_http_proxy').update(
+          value: @default_proxy.name)
+        @repo = katello_repositories(:fedora_17_x86_64)
+
+        RepositorySupport.create_and_sync_repo(@repo)
+      end
+
+      def test_with_global_http_proxy
+        @repo.root.update(http_proxy_policy: RootRepository::GLOBAL_DEFAULT_HTTP_PROXY)
+        RepositorySupport.create_and_sync_repo(@repo)
+      end
+
+      def teardown
+        RepositorySupport.destroy_repo(@repo)
+        User.current = nil
+      end
+    end
```

Comment:
this is more for testing the rpm unit class, i think this test would go better here: https://github.com/Katello/katello/blob/master/test/services/katello/pulp/repository/yum_test.rb